terraform/templates/providers.tmpl

```diff
@@ -0,0 +1,26 @@
+## This file is automatically generated based on what regions are enabled in an AWS account.
+
+%{ for region in account_regions ~}
+provider "aws" {
+  region = "${region}"
+  alias  = "${provider_key}-${region}"
+  %{~ if assume_role }
+  assume_role {
+    role_arn = var.assume_role
+  }
+  %{~ endif ~}
+}
+
+%{ endfor ~}
+
+module "baselines" {
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-baselines"
+  providers = {
+    aws                = aws
+    %{~ for region in account_regions ~}
+    aws.${region} ${substr("              ", 0, 14-length(region))}= aws.${provider_key}-${region}
```

Comment:
This spacing substr is to output the generated file with the correct spacing, so `terraform fmt` doesn't try to reformat the file every time, and also so `terraform plan` doesn't output that the file needs updating to revert `terraform fmt` each time.